### PR TITLE
Adding ability to run the periodic job without test flag.

### PR DESCRIPTION
**Pull request by kwoodson**
Adding periodic build of cluster **without** RUNNING_UNDER_TEST

fixes https://github.com/openshift/openshift-azure/issues/685

### CODE REVIEW

File: ci-operator/templates/openshift/openshift-azure/build-node-image.yaml [hunk 1]
```diff
@@ -25,6 +25,8 @@ parameters:
 - name: DEPLOY_OS
   value: "rhel7"
   required: true
+- name: RUNNING_UNDER_TEST
```
Comment (kwoodson) [inline ci-operator/templates/openshift/openshift-azure/build-node-image.yaml]: @kargakis is this sufficient to allow a parameter to be passed?

Comment (kargakis) [reply ci-operator/templates/openshift/openshift-azure/build-node-image.yaml]: Why are you adding it in the node build job?

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 2]
```diff
@@ -270,6 +270,57 @@ periodics:
           sources:
           - secret:
               name: cluster-secrets-azure
+  - agent: kubernetes
+    cron: ""
+    decorate: true
+    interval: 24h
+    name: azure-build-node-image-rhel-310-not-running-under-test
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: I expected we would merely run tests in openshift-azure w/o specifying RUNNING_UNDER_TEST. Nothing to do with the node build job. I would prefer we continue building images tested w/o RUNNING_UNDER_TEST.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml [hunk 3]
```diff
@@ -272,3 +272,64 @@ presubmits:
...
+      volumes:
+      - name: cluster-secrets-azure
+        projected:
+          sources:
+          - secret:
+              name: cluster-secrets-azure
+      - configMap:
+          name: prow-job-cluster-launch-e2e-azure
+        name: job-definition
+    trigger: ((?m)^/test( all| no-rut),?(\s+|$))
```
Comment (mjudeikis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml]: not sure we want `/test all` here.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 4]
```diff
@@ -279,3 +279,52 @@ periodics:
         requests:
           cpu: 10m
     serviceAccountName: ci-operator
+- agent: kubernetes
+  cron: ""
+  decorate: true
+  interval: 24h
+  name: openshift-azure-not-running-under-test
```
Comment (mjudeikis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: lets keep azure in the front for all jobs, helps to filter easier in prow `azure-test-not-rut`?

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 5]
```diff
@@ -1,6 +1,7 @@
 kind: Template
 apiVersion: template.openshift.io/v1
 
+parameters:
```
Comment (mjudeikis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: typo?

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 6]
```diff
@@ -179,7 +182,7 @@ objects:
         # create cluster for test
         cd /go/src/github.com/openshift/openshift-azure/
         timeout 1h ./hack/create.sh ${RESOURCEGROUP}
-        if [[ -n "$TEST_IN_PRODUCTION" ]]; then
+        if [[ -n "$TEST_IN_PRODUCTION" ]] || [[ "${RUNNING_UNDER_TEST}" == "" ]]; then
           make e2e-prod
```
Comment (mjudeikis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Is this same? @kargakis ?  or we need 3 branches for tests?
E2E - FAKE RP NO RUT
E2E - FAKE RP RUT
E2E - REAL RP?

Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: I had the same question.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml (hunk 3 above)
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml]: will update.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 4 above)
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: makes sense, will update.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 5 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: yeah, will remove.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml [hunk 7]
```diff
@@ -1,5 +1,65 @@
...
+    rerun_command: /test no-rut
+    skip_cloning: true
+    spec:
+      containers:
+      - args:
+        - --artifact-dir=$(ARTIFACTS)
+        - --give-pr-author-access-to-namespace=true
+        - --secret-dir=/usr/local/e2e-secrets/azure
+        - --target=[output:stable:sync]
+        - --target=e2e-azure
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml]: Note that this clashes with the original e2e job in the PR once this runs but I guess that's not a big issue. I don't know of a way to fix this.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 8]
```diff
@@ -179,7 +181,7 @@ objects:
         # create cluster for test
         cd /go/src/github.com/openshift/openshift-azure/
         timeout 1h ./hack/create.sh ${RESOURCEGROUP}
-        if [[ -n "$TEST_IN_PRODUCTION" ]]; then
+        if [[ -n "$TEST_IN_PRODUCTION" ]] || [[ "${RUNNING_UNDER_TEST}" == "" ]]; then
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: This test only makes sense in prod cluster where end users don't have access to azure resources, please remove

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 6 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: @kargakis, if you get a second could you comment on this?

Comment (kargakis) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: This test only makes sense in prod cluster where end users don't have access to azure resources

**Pull request closed without merging**